File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public long getSum(int[] queue){
        int sum = 0;
        for(int i=0; i<queue.length; i++)
            sum += queue[i];
        
        return sum;
    }
    
    public int solution(int[] queue1, int[] queue2) {
        int answer = 0;
        long sum1 = getSum(queue1);
        long sum2 = getSum(queue2);
        Queue<Integer> q1 = new LinkedList<>();
        Queue<Integer> q2 = new LinkedList<>();
        
        for(int i=0; i<queue1.length; i++){
            q1.add(queue1[i]);
            q2.add(queue2[i]);
        }
        
        while(sum1 != sum2){
            if(answer > queue1.length * 4){
                answer = -1;
                break;
            }
            if(sum1 > sum2){
                int x = q1.poll();
                q2.add(x);
                sum1 -= x;
                sum2 += x;
            }
            else{
                int x = q2.poll();
                q1.add(x);
                sum1 += x;
                sum2 -= x;
            }
            
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/lv2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    q1 = deque(queue1)
    q2 = deque(queue2)
    sum1 = sum(q1)
    sum2 = sum(q2)

    while sum1 != sum2:
        if answer > len(queue1)*4:       
            answer = -1
            break
            
        if sum1 > sum2:
            x = q1.popleft()
            q2.append(x)
            sum1 -= x
            sum2 += x
        else:
            x = q2.popleft()
            q1.append(x)
            sum1 += x
            sum2 -= x
        answer += 1
    
    return answer